src/iceberg/catalog/rest/endpoint_util.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
+
+#include <cctype>
+#include <format>
+#include <string>
+#include <string_view>
+
+#include "iceberg/table_identifier.h"
+
+namespace iceberg::rest {
+
+/// \brief Trim a single trailing slash from a URI string_view.
+/// \details If \p uri_sv ends with '/', remove that last character; otherwise the input
+/// is returned unchanged.
+/// \param uri_sv The URI string to trim.
+/// \return The (possibly) trimmed URI string.
+inline std::string TrimTrailingSlash(std::string uri_sv) {
+  if (!uri_sv.empty() && uri_sv.back() == '/') {
+    uri_sv.pop_back();
+  }
+  return uri_sv;
+}
+
+/// \brief Percent-encode a string as a URI component (RFC 3986).
```

Comment:
The Java impl uses `URLEncoder.encode(toEncode, StandardCharsets.UTF_8)`. According to its javadoc, I think the comment here should be `Translates a string into application/x-www-form-urlencoded format using UTF-8`.

Reply:
Are you sure this is `RFC 3986`? I found that the Java references to `RFC 2396`.